test/unit/spec/v2/models/AppState_spec.js

```diff
@@ -111,4 +111,42 @@ describe('v2/models/AppState', function() {
       });
     });
   });
+
+  describe('shouldReRenderView', () => {
+    it('rerender view should be false if stateHandle are different', () => {
+
+      const transformedResponse = {
+        'idx': {
+          'rawIdxState': {
+            'version': '1.0.0',
+            'intent': 'LOGIN',
+            'stateHandle': 'ABCDF',
+            'refresh' : '23',
+            'value': {
```

Comment:
better to mimic real data (the value is nested inside another array)

Reply:
Updated with real payload. (It is trimmed a little bit
)